src/Model/Resolver/Menu.php

```diff
@@ -84,43 +83,48 @@ public function resolve(
         ResolveInfo $info,
         array $value = null,
         array $args = null
-    ): Value {
-        $result = function () {
-            return null;
-        };
-
-        if (isset($args['identifier'])) {
-            $menu = $this->menuCollectionFactory->create();
-
-            /** Updated with identifier filtering */
-            $menu->addFieldToFilter('identifier', $args['identifier'])->load();
+    )
+    {
+        $identifier = $args['identifier'];
 
-            $menuData = $menu->getFirstItem()->getData();
+        $menu = $this->menuFactory->create();
+        $this->menuResourceModel->load($menu, $identifier);
 
-            $items = $this->itemCollectionFactory->create();
-
-            /** Updated with menu id taken from menu data */
-            $items->addMenuFilter($menuData['menu_id'])
-                ->addStatusFilter()
-                ->setParentIdOrder()
-                ->setPositionOrder();
-
-            $menuData['items'] = $items->getData();
+        if ($menu->getId() === null) {
+            throw new \InvalidArgumentException("Could not find menu with identifier '${identifier}'");
```

Comment:
No need for '${}' wrappers, double-quotes allows to use variables directly, without templating, placeholders etc, i.e. "Identifier: $identifier".
However, a better way is to handle string templates using sprintf() function.

Reply:
Changed code to user `sprintf`. Also did for menu item url creation.